File: week5/diamond.cpp
#include<iostream> 
using namespace std; 
class ClassA 
{ 
public: 
    void aClass() 
    { 
        cout<<"Iam a ClassA\n"; 
    } 
}; 
class ClassB: public ClassA 
{ 
public: 
    void bClass() 
    { 
        cout<<"Iam a ClassB\n"; 
    } 
}; 
class ClassC: public ClassA 
{ 
public: 
    void cClass()
    { 
        cout<<"Iam a ClassC\n"; 
    } 
}; 
class ClassD: public ClassC,public ClassB 
{ 
public: 
    void dClass() 
    { 
       cout<<"Iam a ClassD\n"; 
    } 
}; 
int main() 
{ 
ClassD obj1; 
    obj1.bClass(); 
    obj1.cClass(); 
    obj1.dClass(); 
    return 0; 
}

File: week5/diamond.java
class ClassA 
{ 
public 
    void aClass() 
    { 
        System.out.print("I am a ClassA"+"\n"); 
    } 
} 
class ClassB extends ClassA 
{ 
public 
    void bClass() 
    { 
        System.out.print("I am a ClassB"+"\n"); 
    } 
} 
class ClassC extends ClassA 
{ 
public 
    void cClass() 
    { 
        System.out.print("I am a ClassC"+"\n"); 
    }
} 
class DaimondProblem 
{ 
    public static void main(String[] args) 
    { 
        System.out.print("Diamond Problem"+"\n"); 
        ClassB class1=new ClassB(); 
        ClassC class2=new ClassC(); 
        class1.aClass(); 
        class1.bClass(); 
        class2.aClass(); 
        class2.cClass(); 
    } 
}

